woocommerce/woocommerce-android
# Top performers with date period

⚠️ This PR depends on this [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2547)

Closes: #7569

### Description
As part of the new changes we are introducing to support Analytics hub, we were in the need to allow merchants to query top performance products not only by our defined granularities but also by customs date ranges.  This PR aims to allow merchants to query top performance products by custom date ranges, while maintaining the performance improvements introduced [here](https://github.com/woocommerce/woocommerce-android/pull/7411).

### Testing instructions
Log into the app and check the following:

- Top performers are loaded correctly for all stats granularities
- Top performers are invalidated when a new order notification is received. Check for logcat messageInvalidating top performer products after receiving the notification
- Check that after data is invalidated when changing stat granularity, the data is fetched from API and refeshed accordingly with new values.. See the log fetchLeaderboards is displayed in logcat as well as notice how the top performer list is refershed
- Check that once all tabs/options data are loaded at least once, switching between tabs/options doesn't trigger fetching data from API. Basically, check that fetchLeaderboards is never called
- Check that after swipe to refresh is called, then switching between tabs will refresh data fro API

### Images/gif


https://user-images.githubusercontent.com/18119390/197770161-776446ad-170b-42f3-ab6b-f5f2aac7cefd.mp4


https://user-images.githubusercontent.com/18119390/197770194-13b3e5a4-d624-4901-b8a0-f2c71a48d1c9.mp4
